File: clojure/word-count/src/word_count.clj
(ns word-count)

(defn words [phrase]
  (clojure.string/split phrase #"\s+"))

(defn clean [phrase]
  (clojure.string/replace phrase #"[^\w\s]" ""))

(defn word-count [sentence]
  (->> sentence
    (clean)
    (words)
    (map clojure.string/lower-case)
    (frequencies)
  )
)

File: javascript/word-count/word-count.js
var Words = function (){};

Words.prototype.count = function(sentence) {
  var counts = {};
  var words = sentence.trim().split(/\s+/);

  for(var i = 0; i < words.length; i++) {
    word = words[i].toLowerCase()
    if (counts.hasOwnProperty(word)){
      counts[word] += 1;
    } else {
      counts[word] = 1;
    }
  }
  return counts;
}

module.exports = Words
